### PR TITLE
adds users in transaction process and providers to storage items

### PR DESCRIPTION
**Pull request by MaverickRolex**

### CODE REVIEW

File: app/controllers/item_transfers_controller.rb [hunk 1]
```diff
@@ -19,13 +20,15 @@ def create
...
     @transfer = ItemTransfer.find(params[:id])
+    @users = User.all
     @items = Storage.all
     @departments = Department.all
   end
 
   def update
     @transfer = ItemTransfer.find(params[:id])
     @transfer.update_attributes(transfer_params)
+    binding.pry
```
Comment (alexisraca) [inline app/controllers/item_transfers_controller.rb]: aguas

File: app/models/provider.rb [hunk 2]
```diff
@@ -8,15 +8,17 @@ class Provider < ApplicationRecord
...
     while Provider.where(name: self.name).count > 0 do
       self.name = "#{self.name}_1"
     end
   end
 
-  def self.find_by_product_prioritized(item_id)
-    self.joins(storage_providers: :storage).
-      where("storages.id = ?", item_id).order(priority: :asc)
-  end
+  #def self.find_by_product_prioritized(item_id)
```
Comment (alexisraca) [inline app/models/provider.rb]: si no lov as a usar quitalo

File: app/models/storage.rb [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
-  validates :item_name, presence: { message: "The item name field can't be blank" }
-  validates :item_name, uniqueness: { message: "The item has already been registered" }
+  validates :item_name, presence: true
+  validates :item_name, uniqueness: true
 
   has_many :item_transfers, foreign_key: :item_id
   has_many :providers, through: :storage_providers
   has_many :storage_providers
 
+  #scope :find_items_with_providers, -> { joins(storage_providers: :provider) }
```
Comment (alexisraca) [inline app/models/storage.rb]: si no lo vas a usar quitalo

File: app/controllers/item_transfers_controller.rb (hunk 1 above)
Comment (MaverickRolex) [reply app/controllers/item_transfers_controller.rb]: corregido

File: app/models/provider.rb (hunk 2 above)
Comment (MaverickRolex) [reply app/models/provider.rb]: Corregido

File: app/models/storage.rb (hunk 3 above)
Comment (MaverickRolex) [reply app/models/storage.rb]: corregido

**Pull request merged**